### PR TITLE
Added Dijkstra's Algorithm in GoLang

### LINKED ISSUE #535: Dijkstra Algorithm in Golang
This is an open issue for the Dijkstra Algorithm algorithm implementation in Golang. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by android-lewis**
**Fixes #535 

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [👍🏻  ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [👍🏻  ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [👍🏻  ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [👍🏻  ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [👍🏻  ] This pull request shall only be merged once the build passes.

-   [👍🏻  ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
 ## Resources
 
-*   [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
-*   [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
-*   [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
-*   [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
-*   [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
```
Comment (RavinduSachintha) [inline README.md:695]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 2]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
```
Comment (RavinduSachintha) [inline README.md:696]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 3]
```diff
@@ -706,16 +708,19 @@ If you plan to suggest a new algorithm or DS, please make sure to read [the guid
 
 ## Project Maintainers
 
-*   [RavinduSachintha](https://github.com/RavinduSachintha)
-*   [Danushka Herath](https://github.com/danushka96)
-*   [Asitha Indrajith](https://github.com/AsithaIndrajith)
-*   [Sachintha Rathnayake](https://github.com/Sacheerc)
+* [RavinduSachintha](https://github.com/RavinduSachintha)
+* [Danushka Herath](https://github.com/danushka96)
```
Comment (RavinduSachintha) [inline README.md:712]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 4]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
+* [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
+* [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
+* [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
+* [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
+* [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online.
```
Comment (RavinduSachintha) [inline README.md:701]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 5]
```diff
@@ -706,16 +708,19 @@ If you plan to suggest a new algorithm or DS, please make sure to read [the guid
 
 ## Project Maintainers
 
-*   [RavinduSachintha](https://github.com/RavinduSachintha)
-*   [Danushka Herath](https://github.com/danushka96)
-*   [Asitha Indrajith](https://github.com/AsithaIndrajith)
-*   [Sachintha Rathnayake](https://github.com/Sacheerc)
+* [RavinduSachintha](https://github.com/RavinduSachintha)
```
Comment (RavinduSachintha) [inline README.md:711]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 6]
```diff
@@ -706,16 +708,19 @@ If you plan to suggest a new algorithm or DS, please make sure to read [the guid
...
+* [Danushka Herath](https://github.com/danushka96)
+* [Asitha Indrajith](https://github.com/AsithaIndrajith)
+* [Sachintha Rathnayake](https://github.com/Sacheerc)
```
Comment (RavinduSachintha) [inline README.md:714]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 7]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
-*   [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
-*   [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
-*   [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
-*   [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
+* [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
+* [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
```
Comment (RavinduSachintha) [inline README.md:698]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 8]
```diff
@@ -706,16 +708,19 @@ If you plan to suggest a new algorithm or DS, please make sure to read [the guid
 
 ## Project Maintainers
 
-*   [RavinduSachintha](https://github.com/RavinduSachintha)
-*   [Danushka Herath](https://github.com/danushka96)
-*   [Asitha Indrajith](https://github.com/AsithaIndrajith)
-*   [Sachintha Rathnayake](https://github.com/Sacheerc)
+* [RavinduSachintha](https://github.com/RavinduSachintha)
+* [Danushka Herath](https://github.com/danushka96)
+* [Asitha Indrajith](https://github.com/AsithaIndrajith)
```
Comment (RavinduSachintha) [inline README.md:713]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 9]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
-*   [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
-*   [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
+* [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
+* [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
+* [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
+* [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
```
Comment (RavinduSachintha) [inline README.md:700]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 10]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
-*   [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
-*   [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
-*   [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
+* [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
+* [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
+* [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
```
Comment (RavinduSachintha) [inline README.md:699]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md [hunk 11]
```diff
@@ -653,50 +655,50 @@ The community maintained a list of **Algorithms and Data Structures** implementa
...
-*   [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
-*   [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
-*   [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
-*   [Intro to Algorithms - Khan Academy](https://www.khanacademy.org/computing/computer-science/algorithms)
-*   [Popular Data Structures and Algorithms - Codechef](https://discuss.codechef.com/questions/48877/data-structures-and-algorithms)
-*   [Stanford-ACM-Codes](https://github.com/jaehyunp/stanfordacm) - A list of codes written by previous Stanford ACM team members and coaches.
-*   [Data Structures and Algorithms](https://hackr.io/tutorials/learn-data-structures-algorithms) - A user ranked list of online tutorials to learn Data Structures and Algorithms online. 
+* [Algorithms - Learneroo](https://www.learneroo.com/subjects/8)
+* [Awesome-Algorithms](https://github.com/tayllan/awesome-algorithms)
+* [Algorithms List - GeeksforGeeks](http://www.geeksforgeeks.org/fundamentals-of-algorithms/)
```
Comment (RavinduSachintha) [inline README.md:697]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

**Review by RavinduSachintha — CHANGES_REQUESTED** [review]
Please don't do any unnecessary changes in the readme file. only need to put the right mark at the right place. :+1:

File: README.md (hunk 8 above)
Comment (RavinduSachintha) [inline README.md:713]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 6 above)
Comment (RavinduSachintha) [inline README.md:714]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 11 above)
Comment (RavinduSachintha) [inline README.md:697]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 5 above)
Comment (RavinduSachintha) [inline README.md:711]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 9 above)
Comment (RavinduSachintha) [inline README.md:700]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 4 above)
Comment (RavinduSachintha) [inline README.md:701]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 3 above)
Comment (RavinduSachintha) [inline README.md:712]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 10 above)
Comment (RavinduSachintha) [inline README.md:699]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 2 above)
Comment (RavinduSachintha) [inline README.md:696]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 1 above)
Comment (RavinduSachintha) [inline README.md:695]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

File: README.md (hunk 7 above)
Comment (RavinduSachintha) [inline README.md:698]: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6333126)

**Comment by android-lewis** [comment]
> Please don't do any unnecessary changes in the readme file. only need to put the right mark at the right place. :+1: 

I'll fix that now

**Comment by android-lewis** [comment]
Updated README

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@all-contributors please add @android-lewis for code and doc

**Comment by RavinduSachintha** [comment]
@android-lewis good job :+1: keep it up.
and make sure to star this repository. :wink:

**Pull request merged**